packages/circuits/circuits/leafHasher.circom

```diff
@@ -61,10 +63,13 @@ template StateTreeLeaf(FIELD_COUNT) {
     hasher[x].inputs[1] <== data[x+1];
   }
 
+  control <== attester_id + 2**160*epoch;
+  data_hash <== hasher[FIELD_COUNT-2].out;
+
   component final_hasher = Poseidon(3);
   final_hasher.inputs[0] <== identity_secret;
-  final_hasher.inputs[1] <== attester_id + 2**160*epoch;
-  final_hasher.inputs[2] <== hasher[FIELD_COUNT-2].out;
+  final_hasher.inputs[1] <== control;
+  final_hasher.inputs[2] <== data_hash;
```

Comment:
I think we should adjust the state tree leaf so it's like

```
H(H(identitySecret, epoch, attesterId), H(data))
```

We can call the `H(identitySecret, epoch, attesterId)` the state leaf identity hash, and `H(data)` the state leaf data hash.

This way an application can provide the state leaf identity hash, and calculate the data hash onchain to build the state leaf.